src/shared/components/QueryBuilder/domain/helpers/queryToFilters.ts

```diff
@@ -27,42 +30,40 @@ export function queryToFilters(query: string): Filters {
   return (rawFilters as string[][])
     .filter(([attribute]) => attribute !== 'service_name')
     .map(([attribute, operator, value]) => {
-      const shouldChangeToIsEmptyOperator = operator === OperatorKind['='] && value === '';
-      if (shouldChangeToIsEmptyOperator) {
-        return {
-          id: nanoid(10),
-          active: true,
-          attribute: { value: attribute, label: attribute },
-          ...IsEmptyFilter,
-        };
-      }
-
-      // TODO: uncomment when we'll support the "in" operator
-      // const shouldChangeToInOperator = operator === OperatorKind['=~'] && value.includes('|');
-      // if (shouldChangeToInOperator) {
-      //   return {
-      //     id: nanoid(10),
-      //     type: FilterKind['attribute-operator-value'],
-      //     active: true,
-      //     attribute: { value: attribute, label: attribute },
-      //     operator: { value: OperatorKind.in, label: OperatorKind.in },
-      //     value: {
-      //       value: value,
-      //       label: value
-      //         .split('|')
-      //         .map((v) => v.trim())
-      //         .join(', '),
-      //     },
-      //   };
-      // }
-
-      return {
+      const filter = {
         id: nanoid(10),
         type: FilterKind['attribute-operator-value'],
         active: true,
         attribute: { value: attribute, label: attribute },
         operator: { value: operator, label: operator },
         value: { value: value, label: value },
       };
+
+      const shouldConvertToIsEmptyOperator = operator === OperatorKind['='] && value === '';
+      if (shouldConvertToIsEmptyOperator) {
+        return buildIsEmptyFilter(filter);
+      }
+
+      const shouldConvertToInNotInOperator =
+        [OperatorKind['=~'], OperatorKind['!~']].includes(operator as OperatorKind) && !REGEX_CHARS_REGEX.test(value);
```

Comment:
To address https://github.com/grafana/explore-profiles/pull/122#issuecomment-2293671380